include/ConformalTracking.h

```diff
@@ -64,44 +64,44 @@ class ConformalTracking : public Processor {
   void getCollection(LCCollection*&, std::string, LCEvent*);
 
   // Plotting function for displaying cells
-  void drawline(KDCluster*, KDCluster*, int, int style = 1);
+  void drawline(SKDCluster const&, SKDCluster const&, int, int style = 1);
 
   // Pattern recognition algorithms:
 
   // Cell creation
-  KDCluster* extrapolateCell(Cell*, double);
-  void       extendSeedCells(std::vector<Cell*>&, KDTree*, bool, const std::vector<KDCluster*>&);
+  SKDCluster extrapolateCell(Cell::SCell const&, double);
+  void       extendSeedCells(SharedCells&, KDTree*, bool, const SharedKDClusters&);
 
   // Track finding
-  void buildNewTracks(std::vector<KDTrack*>&, std::vector<KDCluster*>&, KDTree*, bool radialSearch = false);
-  void extendTracks(std::vector<KDTrack*>&, std::vector<KDCluster*>&, KDTree*);
-  void combineCollections(std::vector<KDCluster*>&, KDTree*&, std::vector<int>, std::map<int, std::vector<KDCluster*>>);
+  void buildNewTracks(UniqueKDTracks&, SharedKDClusters&, KDTree*, bool radialSearch = false);
```

Comment:
Should one also make the KDTree a unique/shared pointer ?

Reply:
Done